Sources/Extensions/UIKit/UIEdgeInsetsExtensions.swift

```diff
@@ -44,6 +44,29 @@ extension UIEdgeInsets {
     public init(horizontal: CGFloat, vertical: CGFloat) {
         self.init(top: vertical/2, left: horizontal/2, bottom: vertical/2, right: horizontal/2)
     }
+
+    /// SwifterSwift: Creates an `UIEdgeInsets` based on current value and offsets.
+    ///
+    /// - Parameters:
+    ///   - top: Offset to be applied in to the top edge.
+    ///   - left: Offset to be applied in to the left edge.
+    ///   - bottom: Offset to be applied in to the bottom edge.
+    ///   - right: Offset to be applied in to the right edge.
+    /// - Returns: UIEdgeInsets offset with given offsets.
+    public func insetBy(top: CGFloat = 0, left: CGFloat = 0, bottom: CGFloat = 0, right: CGFloat = 0) -> UIEdgeInsets {
```

Comment:
Giving default values for all parameter allow the user to call insets.insetBy() with no parameters. I'm not sure if it a viable option for an API standpoint. @VincentSit  What are your thoughts about it?

Reply:
I have also noticed the problem.

If do not add a default value, it is not necessary to invoke the method to pass all the values. This is probably the easiest way to do this, and my other idea is to add an edge option set and use it as an argument. ie:

```
public enum Edges: OptionSet {
    case top(CGFloat)
    case left(CGFloat)
    case bottom(CGFloat)
    case right(CGFloat)
    case horizontal(CGFloat)
    case vertical(CGFloat)
    case all(CGFloat)
}

public func insetBy(edge: Edges) -> UIEdgeInsets {}

let insets = UIEdgeInsets(top: 10, left: 15, bottom: 20, right: 25)
view.frame.inset(by: insets.insetBy(.bottom(5)))
view.frame.inset(by: insets.insetBy([.bottom(5), .right(10)]))
```

Reply:
In theory, the parameter( `edges` in `insetBy(edges:)`) should use `OptionSet`, but I tried to find that `OptionSet` and the Enum case with parameters do not work well together. 

Currently I have two options:
1. Use an Edge array if you think that only one case is also needed to use an array is acceptable. E.g. `[.top(10)]`, `[.top(10), .left(15)]`.
2. Split into separate functions. E.g. `insetBy(top:)`, `insetBy(left:)`, `insetBy(bottom:)`...

If there is no better solution, it is okay to reject this PR.

Reply:
Hey @VincentSit 
Seems a viable solution, but I believe that the fact it requires creating a new type makes it not fit well with this project :) 
I was thinking that if you only define only the four methods `insetBy(top:)`, `insetBy(left:)`, `insetBy(bottom:)` and `insetBy(right:)`. We can still have the abstraction, and from the API standpoint the user will be able to compose the call like
```swift
let insets = UIEdgeInsets(top: 10, left: 15, bottom: 20, right: 25)
view.frame.inset(by: insets.insetBy(bottom:5))
// or composing
view.frame.inset(by: insets.insetBy(bottom:5)
                                             .insetBy(right: 5))
view.frame.inset(by: insets.insetBy(bottom: 5)
                                             .insetBy(horizontal: 10))
```
I believe it's an option 👍 What you think?
No need to reject the PR, we can start a good discussion here and see if we can come up with more options :))

Reply:
Hi @LeLuckyVint 
I understand what you mean and agree with your suggestion. But if you only define the four methods you mentioned, `insetBy(horizontal:)` doesn't seem to be able to be called.

Reply:
Hey @VincentSit I mention the 4 just as example, my idea was all of them follow the same pattern thats why I put the horizontal in the code sample :))

Reply:
Hi @LucianoPAlmeida I have updated the code. Please review and give your opinion when you are free.